src/main/java/com/google/ddexmeadparser/DdexMeadParser.java

```diff
@@ -0,0 +1,258 @@
+    logger.atInfo().log(
+        "Started mead message CONVERSION on file: " + runtimeOptions.inputFile.getName());
+    Document document = getDocument(runtimeOptions.inputFile);
+
+    MeadConverter meadConverter = new MeadConverter();
+    Message message = meadConverter.convert(document);
+
+    // Write output proto message files
+    logger.atInfo().log(message.toString());
+  }
+
+  /**
+   * Parse schema.
+   *
+   * @throws SchemaConversionException the schema conversion exception
+   * @throws InvalidOptionsException the invalid options exception
+   */
+  public void parseSchema() throws SchemaConversionException, InvalidOptionsException {
+    logger.atInfo().log("Started schema PARSE on file: " + runtimeOptions.inputFile.getName());
+    StreamSource xsdFile = getStreamSource(runtimeOptions.inputFile);
+
+    SchemaConverter schemaConverter = new SchemaConverter();
+    SchemaEntryMap schemaEntryMap = schemaConverter.convert(xsdFile);
+
+    ProtoSchema protoSchema = new ProtoSchema(schemaEntryMap);
+
+    // Write schema to file
+    writeSchema(protoSchema);
+  }
+
+  private void writeSchema(ProtoSchema schema) throws SchemaConversionException {
+    String rootNamespace = schema.getRootNamespace();
+    String packageName = schema.getPackageName();
+
+    Set<String> namespaces = schema.getSchemaStringMap().keySet();
+    for (String namespace : namespaces) {
+      File file =
+          new File(
+              "./src/main/proto/" + rootNamespace + "/" + packageName + "/" + namespace + ".proto");
+      file.getParentFile().mkdirs();
+
+      try {
+        FileWriter writer = new FileWriter(file, false);
+        writer.write(schema.getSchemaStringMap().get(namespace));
+        writer.close();
+      } catch (IOException e) {
+        throw new SchemaConversionException("Could not write schema to file.", e);
+      }
+    }
+  }
+
+  private void parseSchemaSet() throws InvalidOptionsException, SchemaConversionException {
+    assert (runtimeOptions.inputFile.isDirectory());
+
+    SchemaSetMerger schemaSetMerger = new SchemaSetMerger();
+
+    for (File schemaFile : runtimeOptions.inputFileList) {
+      logger.atInfo().log("Started schema set PARSE on folder: " + schemaFile.getName());
+      StreamSource xsdFile = getStreamSource(schemaFile);
+
+      SchemaConverter schemaConverter = new SchemaConverter();
+      SchemaEntryMap schemaEntryMap = schemaConverter.convert(xsdFile);
+      schemaSetMerger.addSchema(schemaEntryMap);
+    }
+
+    SchemaEntryMap finalMap = schemaSetMerger.merge();
+    ProtoSchema protoSchema = new ProtoSchema(finalMap);
+
+    // Write schema to file
+    writeSchema(protoSchema);
+  }
+
+  private Document getDocument(File file) throws InvalidOptionsException {
+    try {
+      if (!file.exists() || file.isDirectory()) {
+        throw new FileNotFoundException("XML file input does not exist or is a directory.");
+      }
```

Comment:
How about moving this if block out of the try{}?
e.g. 
---- early return first
if (!file.exists() || file.isDirectory()) {
    throw new FileNotFoundException("XML file input does not exist or is a directory.");
}
try {
  ....
} catch ...

It's easier to read with an early return(less nested brackets). Also, it's more clear that the ParserConfigurationException, IOException, and SAXException are from the `DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(file)`